test/sys/test_prctl.rs

```diff
@@ -0,0 +1,106 @@
+use nix::sys::prctl;
+
+#[cfg_attr(qemu, ignore)]
+#[test]
+fn test_get_set_subreaper() {
+    let subreaper = prctl::get_child_subreaper().unwrap();
+    assert_eq!(subreaper, false);
+
+    prctl::set_child_subreaper(true).unwrap();
+    let subreaper = prctl::get_child_subreaper().unwrap();
+    assert_eq!(subreaper, true);
+}
+
+#[test]
+fn test_get_set_dumpable() {
+    let dumpable = prctl::get_dumpable().unwrap();
+    assert_eq!(dumpable, true);
+
+    prctl::set_dumpable(false).unwrap();
+    let dumpable = prctl::get_dumpable().unwrap();
+    assert_eq!(dumpable, false);
+}
+
+#[test]
+fn test_get_set_keepcaps() {
+    let keepcaps = prctl::get_keepcaps().unwrap();
+    assert_eq!(keepcaps, false);
+
+    prctl::set_keepcaps(true).unwrap();
+    let keepcaps = prctl::get_keepcaps().unwrap();
+    assert_eq!(keepcaps, true);
+}
+
+#[test]
+fn test_get_set_clear_mce_kill() {
+    use prctl::PrctlMCEKillPolicy::*;
+
+    prctl::set_mce_kill(PR_MCE_KILL_LATE).unwrap();
+    let mce = prctl::get_mce_kill().unwrap();
+    assert_eq!(mce, PR_MCE_KILL_LATE);
+
+    prctl::clear_mce_kill().unwrap();
+    let mce = prctl::get_mce_kill().unwrap();
+    assert_eq!(mce, PR_MCE_KILL_DEFAULT);
+}
+
+#[cfg_attr(qemu, ignore)]
+#[test]
+fn test_get_set_pdeathsig() {
+    use nix::sys::signal::Signal;
+
+    let sig = prctl::get_pdeathsig().unwrap();
+    assert_eq!(sig, None);
+
+    prctl::set_pdeathsig(Signal::SIGUSR1).unwrap();
+    let sig = prctl::get_pdeathsig().unwrap();
+    assert_eq!(sig, Some(Signal::SIGUSR1));
+}
+
+#[test]
+fn test_get_set_name() {
+    let long_name = String::from("0123456789abcdefghijklmn");
+    prctl::set_name(&long_name).unwrap();
+    let res = prctl::get_name().unwrap();
+    assert_eq!(&long_name[..15], res); // name truncated by kernel
+
+    let short_name = String::from("01234567");
+    prctl::set_name(&short_name).unwrap();
+    let res = prctl::get_name().unwrap();
+    assert_eq!(short_name, res);
+}
+
+#[cfg_attr(qemu, ignore)]
+#[test]
+fn test_get_set_timerslack() {
+    let slack = 60_000;
+    prctl::set_timerslack(slack).unwrap();
+    let res = prctl::get_timerslack().unwrap();
+    assert_eq!(slack, res as u64);
+}
+
+#[test]
+fn test_disable_enable_perf_events() {
+    prctl::task_perf_events_disable().unwrap();
+    prctl::task_perf_events_enable().unwrap();
+}
+
+#[test]
+fn test_get_set_no_new_privs() {
+    let res = prctl::get_no_new_privs().unwrap();
+    assert_eq!(res, false);
+
+    prctl::set_no_new_privs().unwrap();
+    let res = prctl::get_no_new_privs().unwrap();
+    assert_eq!(res, true);
+}
```

Comment:
Needs cleanup

Reply:
This one can't easily be cleaned up. It is an irreversible operation. It acts on the thread so I don't think it should affect other tests.

Reply:
But if `--test-threads=1`, then Rust's test harness doesn't create a new thread per test.  It just reuses one.  So it might affect other tests.  I suggest that you either explicitly spawn a thread for this test, or else create its own test program.